File: longest-repeating-character-replacement/longest-repeating-character-replacement.java
class Solution {
    public int characterReplacement(String s, int k) {

        char[] charArr = new char[26];

        int i = 0, longest = 0, modeCharCount = 0;

        for (int j = 0; j < s.length(); j++) {
            modeCharCount = Math.max(modeCharCount, ++charArr[s.charAt(j) - 'A']);
            while (j - i + 1 - modeCharCount > k) {
                charArr[s.charAt(i) - 'A']--;
                i++;
            }
            longest = Math.max(longest, j - i + 1);
        }

        return longest;
    }
}

File: longest-repeating-character-replacement/longest-repeating-character-replacement.py
from collections import Counter
class Solution:
    def characterReplacement(self, s: str, k: int) -> int:
        
        
        chars = Counter()
        i = j = longest = 0
        while j < len(s):
            chars[s[j]] += 1
            if j - i + 1 - max(chars.values()) > k:
                chars[s[i]] -= 1
                i+=1
            longest = max(longest, j-i+1)
            j+=1
        
        return longest